src/overlays/effects/ovl_Effect_Ss_Solder_Srch_Ball/z_eff_ss_solder_srch_ball.c

```diff
@@ -1,6 +1,55 @@
-#include <ultra64.h>
-#include <global.h>
+/*
+ * File: z_eff_ss_solder_srch_ball.c
+ * Overlay: ovl_Effect_Ss_Solder_Srch_Ball
+ * Description: Vision sphere for courtyard guards
+ */
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/effects/ovl_Effect_Ss_Solder_Srch_Ball/func_809AC130.s")
+#include "z_eff_ss_solder_srch_ball.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/effects/ovl_Effect_Ss_Solder_Srch_Ball/func_809AC1AC.s")
+u32 EffSsSolderSrchBall_Init(GlobalContext* globalCtx, u32 index, EffectSs* this,
+                             EffSsSolderSrchBallInitParams* initParams);
+void EffSsSolderSrchBall_Update(GlobalContext* globalCtx, u32 index, EffectSs* this);
+
+EffectSsInit Effect_Ss_Solder_Srch_Ball_InitVars = {
+    EFFECT_SS_SOLDER_SRCH_BALL,
+    EffSsSolderSrchBall_Init,
+};
+
+u32 EffSsSolderSrchBall_Init(GlobalContext* globalCtx, u32 index, EffectSs* this,
+                             EffSsSolderSrchBallInitParams* initParams) {
```

Comment:
Since this is an entrypoint, the init params should most likely be generic (a `void*`, which you'll notice I used in the `EffectSsInit` struct), and casted inside the system as usual.

For other effects I have been calling it `initParamsx` to stay consistent with the `thisx` pattern.
```suggestion
                             void* initParamsx) {
```